apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -149,10 +150,13 @@ private void publishEvent(
     }
     if (partition != null) {
       JSONArray partitionValuesArray = new JSONArray(partition.getValues());
-      JSONArray partitionKeysArray = new JSONArray(
-          table.getPartitionKeys().stream().map(f -> f.getName()).collect(Collectors.toList()));
+      Map<String, String> partitionKeysMap = new LinkedHashMap<>();
+      for (FieldSchema fieldSchema : partition.getSd().getCols()) {
+        partitionKeysMap.put(fieldSchema.getName(), fieldSchema.getType());
```

Comment:
IIRC `fieldSchema.getType()` always returns all lower case values which is why the examples generated in the CT SNS module are all lower case as it uses this same getter to get the type values. So as long as the examples above are corrected then this is fine.